lib/stub_alloc_pages.c

```diff
@@ -42,6 +42,15 @@
 #include <caml/fail.h>
 #include <caml/bigarray.h>
 
+/* Expose the address of the bigarray (in other words, leak the pointer).
+  */
+CAMLprim value
+caml_address(value t)
+{
+  long data = (long)Caml_ba_data_val(t) ;
+  return Val_long(data);
```

Comment:
A couple of problems here:
* a `long` can't necessarily hold a pointer (e.g. on [platforms](https://github.com/ocaml/ocaml/blob/472ee2c0ff40ce1daa5afb27cf2077e2df83590e/byterun/caml/config.h#L91) where pointers are 64 bits and longs 32)
* The `external` declaration says that this returns a `nativeint`, so the return value should be created with [`caml_copy_nativeint`](https://github.com/ocaml/ocaml/blob/472ee2c0ff40ce1daa5afb27cf2077e2df83590e/byterun/ints.c#L703-L708), not `Val_long`